### PR TITLE
Add condition to check failed state for analysis

### PR DESCRIPTION
**Pull request by gabriel-farache**
This PR adds a condition to check for failed state of the current taskGroup to avoid endless loop if the analysis failed

### CODE REVIEW

File: mta/mta.sw.yaml [hunk 1]
```diff
@@ -187,5 +189,28 @@ states:
...
+        functionRef:
+          refName: createNotification
+          arguments:
+            title: "MTA analysis failed"
+            message: '"MTA analysis for " + .application.repository.url + " failed. Task pod: " + .taskgroup.pod +" ; Error: "+ .taskgroup.error'
+            origin: "MTA Analysis Workflow"
+            topic: "MTA Analysis Workflow"
+            targetGroups:
+              - .backstageGroup
+    end: true
```
Comment (rgolangh) [inline mta/mta.sw.yaml:215]: the ui plugin can deal with an assessment returning no options structue or we need an empty structure?

Comment (gabriel-farache) [reply mta/mta.sw.yaml:215]: Good question, I'll test without sending notification to have the workflow reach the end successfully

File: mta/mta.sw.yaml [hunk 2]
```diff
@@ -187,5 +189,28 @@ states:
...
+        functionRef:
+          refName: createNotification
+          arguments:
+            title: "MTA analysis failed"
+            message: '"MTA analysis for " + .application.repository.url + " failed. Task pod: " + .taskgroup.pod +" ; Error: "+ .taskgroup.error'
+            origin: "MTA Analysis Workflow"
+            topic: "MTA Analysis Workflow"
+            targetUsers:
+              - .backstageUser
+      - condition: .backstageGroup != null
```
Comment (rgolangh) [inline mta/mta.sw.yaml:205]: I think we can shrink this to a single operation where we send both targetGroups and targetUsers arguments:
```
condition: .backstageUser != null || backstageGroup != null
...
targetUsers:
 - if .backstageUser != null  then .backstageUser else "" end
 targetGroups:
 - if .backstageGroup != null then .targetGroup else "" end 
```

I really hope the notification API can work with empty/nulls. I think it was better if the API could take bot users and group uuids. different discussion though

**Review by rgolangh — COMMENTED** [review]
Give a try to reducing the code to  asingle operation for notification. If its not working because of the notification API then I'll approve this with this structure for now. Although I think this verbosity is bad

File: mta/mta.sw.yaml (hunk 2 above)
Comment (gabriel-farache) [reply mta/mta.sw.yaml:205]: I'll test to set the target to empty string or null and see how the notifications plugin react to that

Comment (gabriel-farache) [reply mta/mta.sw.yaml:205]: so, no it does not work, if empty or null is given, the notification plugin does not like it

File: mta/mta.sw.yaml (hunk 1 above)
Comment (gabriel-farache) [reply mta/mta.sw.yaml:215]: The system not care if there is not output, the state is completed and no link is displayed
image

**Comment by gabriel-farache** [comment]
> Give a try to reducing the code to asingle operation for notification. If its not working because of the notification API then I'll approve this with this structure for now. Although I think this verbosity is bad

@rgolangh so, the notification API is not flexible so I cannot merge the 2 conditions
I agree that duplicating code is bad but in this case, unfortunately we have no choice, if the API evolves then we can revisit

**Review by masayag — APPROVED** [review]

**Pull request merged**